### PR TITLE
Add content-type header in fileserver

### LINKED ISSUE #43: file_server should serve Content-Type header for common file formats

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes #43. 

I took list of common mime types from [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Complete_list_of_MIME_types) and extended it with `.yml` and `.yaml` formats so I could reuse already existing tests. 

If extension if not recognized it falls back to `application/octet-stream` which causes file to be downloaded by default. IMHO we could extended list more and provide it as separate module, then serve more files as plaintext instead of octet stream.

### CODE REVIEW

File: net/file_server.ts [hunk 1]
```diff
@@ -12,6 +12,81 @@ import {
...
+  ["", "application/octet-stream"],
+  [".aac", "audio/aac"],
+  [".abw", "application/x-abiword"],
+  [".arc", "application/octet-stream"],
+  [".avi", "video/x-msvideo"],
+  [".azw", "application/vnd.amazon.ebook"],
+  [".bin", "application/octet-stream"],
+  [".bmp", "image/bmp"],
+  [".bz", "application/x-bzip"],
+  [".bz", "application/x-bzip2"],
```
Comment (hayd) [inline net/file_server.ts]: bz2?

Comment (bartlomieju) [reply net/file_server.ts]: Good catch, thanks

File: net/file_server.ts [hunk 2]
```diff
@@ -12,6 +12,81 @@ import {
   Response
 } from "./http.ts";
 import { cwd, DenoError, ErrorKind, args, stat, readDir, open } from "deno";
+import { extname } from "https://deno.land/x/path/index.ts";
+
+// based on https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Complete_list_of_MIME_types
+const extensionsMap = new Map([
```
Comment (kevinkassimo) [inline net/file_server.ts]: Probably we could move this to another file (or even to a separate module)? It is possible that later other modules might also need such mappings...

Comment (ry) [reply net/file_server.ts]: Having it in a JSON file would be good

File: net/file_server.ts [hunk 3]
```diff
@@ -12,6 +12,8 @@ import {
   Response
 } from "./http.ts";
 import { cwd, DenoError, ErrorKind, args, stat, readDir, open } from "deno";
+import { extname } from "https://deno.land/x/path/index.ts";
```
Comment (ry) [inline net/file_server.ts]: Use relative paths within `deno_std` so that if people link to a specific version they will get a compatible version of paths.

Comment (bartlomieju) [reply net/file_server.ts]: Yeah I wasn't sure about that, so went with registry, I'll open up another PR for that

**Pull request merged**